pf4j/src/main/java/org/pf4j/LegacyExtensionFinder.java

```diff
@@ -83,14 +83,21 @@ public LegacyExtensionFinder(PluginManager pluginManager) {
             Set<String> bucket = new HashSet<>();
 
             try {
-                URL url = ((PluginClassLoader) plugin.getPluginClassLoader()).findResource(getExtensionsResource());
-                if (url != null) {
-                    log.debug("Read '{}'", url.getFile());
-                    try (Reader reader = new InputStreamReader(url.openStream(), StandardCharsets.UTF_8)) {
-                        LegacyExtensionStorage.read(reader, bucket);
-                    }
-                } else {
+                Enumeration<URL> urls = ((PluginClassLoader) plugin.getPluginClassLoader()).findResources(getExtensionsResource());
+                if (urls == null || !urls.hasMoreElements()) {
                     log.debug("Cannot find '{}'", getExtensionsResource());
+                } else {
+                    while (urls.hasMoreElements()) {
+                        URL url = urls.nextElement();
+                        if (url != null) {
```

Comment:
Url is not null, so no need for null check

Reply:
You're right. This null check is definitely not necessary. I've [updated the pull request](https://github.com/pf4j/pf4j/pull/262/commits/23a4263eef7de59ed4a5bb29f84af4010b749d75) accordingly.